src/compiler/crystal/macros/methods.cr

```diff
@@ -947,6 +953,16 @@ module Crystal
   class NamedTupleLiteral
     def interpret(method : String, args : Array(ASTNode), named_args : Hash(String, ASTNode)?, block : Crystal::Block?, interpreter : Crystal::MacroInterpreter)
       case method
+      when "delete"
+        interpret_one_arg_method(method, args) do |key|
+          unless key.is_a? Crystal::SymbolLiteral | Crystal::StringLiteral
```

Comment:
I'm not sure that `|` falls inside the `is_a?`. Try using parentheses. Or doing `key.is_a?(...) || key.is_a?(...)`.